student-hub-app/src/Home/SignUp.js

```diff
@@ -0,0 +1,68 @@
+// SignUp.js
+
+import React, { useState } from 'react';
+
+const SignUp = () => {
+    const [formData, setFormData] = useState({
+        username: '',
+        email: '',
+        password: '',
+    });
+
+    const handleChange = (e) => {
+        setFormData({ ...formData, [e.target.name]: e.target.value });
+    };
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        // Perform signup logic here, e.g., make an API call
+        console.log('SignUp Form Data:', formData);
+        // Reset form after submission
+        setFormData({
+            username: '',
+            email: '',
+            password: '',
+        });
+    };
+
+    return (
+        <div>
+            <h2>Sign Up</h2>
+            <form onSubmit={handleSubmit}>
+                <label>
+                    Username:
+                    <input
+                        type="text"
+                        name="username"
+                        value={formData.username}
+                        onChange={handleChange}
+                    />
+                </label>
+                <br />
+                <label>
+                    Email:
+                    <input
+                        type="email"
+                        name="email"
+                        value={formData.email}
+                        onChange={handleChange}
+                    />
+                </label>
+                <br />
+                <label>
+                    Password:
+                    <input
+                        type="password"
```

Comment:
We should put a confirm password box here too

Reply:
^